components/file-selector.vue

```diff
@@ -1,152 +1,159 @@
+                
+                knownContentTypes: {
 
-				fileInput: null
-			}
-		},
+                    "json": "application/json",
+                    "tsv": "text/tab-separated-values"
+                },
 
-		computed: {
+                fileInput: null,
 
-			fileName() {
+                uiText: {
 
-				return ( null == this.fileInput ) ? "" : this.fileInput.name;
-			}
-		},
+                    instructions: "Choose file"
+                }
+            }
+        },
 
-		methods: {
+        computed: {
 
-			onFileSelected(p_event){
+            fileName() {
 
-				// 1. Save the file name
-				this.fileInput = p_event.target.files[0];
+                return ( null === this.fileInput ) ? "" : this.fileInput.name;
+            }
+        },
 
-				// A. If no file selected, emit a no file selected message
-				if ( "undefined" == typeof this.fileInput ) {
-					this.$emit("file-selected", "none");
-					return;
-				}
-				
-				// 2. Parse the whole file and save the lines
-				
-				// A. TSV file parsing
-				if ( this.knownContentTypes["tsv"] == this.contentType ) {
+        methods: {
 
-					Papa.parse(this.fileInput, {
+            onFileSelected(p_event){
 
-						complete: results => {
+                // 1. Save the file name
+                this.fileInput = p_event.target.files[0];
 
-							// I. Send the file data to the store to be processed and saved
-							this.$emit("file-selected", results.data);
-						},
-					});
-				} 
-				// B. JSON file parsing
-				else if ( this.knownContentTypes["json"] == this.contentType ) {
+                // A. If no file selected, emit a no file selected message
+                if ( "undefined" === typeof this.fileInput ) {
+                    this.$emit("file-selected", "none");
+                    return;
+                }
+                
+                // 2. Parse the whole file and save the lines
+                
+                // A. TSV file parsing
+                if ( this.knownContentTypes["tsv"] === this.contentType ) {
 
-					// I. Reference to this json object in this component's data
-					var myJson;
+                    Papa.parse(this.fileInput, {
 
-					// II. Create a file reader object for reading the json file contents
-					let reader = new FileReader();
+                        complete: results => {
 
-					// III. On loading the file contents:
-					reader.onload = e => {
+                            // I. Send the file data to the store to be processed and saved
+                            this.$emit("file-selected", results.data);
+                        },
+                    });
+                } 
+                // B. JSON file parsing
+                else if ( this.knownContentTypes["json"] === this.contentType ) {
 
-						// a. Save a reference to the loaded contents
-						myJson = e.target.result;
+                    // I. Reference to this json object in this component's data
+                    var myJson;
```

Comment:
since you're making these things consistent, I'd use 
```suggestion
                    let myJson;
```
here